recipe/recipe.yaml

```diff
@@ -0,0 +1,145 @@
+  ignore_run_exports:
+    by_name:
+      - cuda-version
+    from_package:
+      - if: match(cuda_compiler_version, "11.*")
+        then:
+          - cudatoolkit
+      - if: match(cuda_compiler_version, "12.*")
+        then:
+          - libcufft-dev
+          - libcufile-dev
+          - libcurand-dev
+          - libnpp-dev
+          - libnvjpeg-dev
+      - libnvimgcodec-dev  ${{ libnvimgcodec }}
+  build:
+    - ${{ compiler('cuda') }}
+    - ${{ compiler('cxx') }}
+    - ${{ stdlib('c') }}
+    - cmake 3.*
+    - file
+    - ninja
+    - if: build_platform != target_platform
+      then:
+        - cross-python_${{ target_platform }}
+        - python
+        - python-clang
+  host:
+    - if: match(cuda_compiler_version, "11.*")
+      then:
+        - cudatoolkit
+    - if: match(cuda_compiler_version, "12.*")
+      then:
+        - cuda-cudart-static
+        - cuda-nvml-dev
+        - libcufft-dev
+        - libcufile-dev
+        - libcurand-dev
+        - libnpp-dev
+        - libnvjpeg-dev
+    - cuda-version ${{ cuda_compiler_version }}.*
+    - cfitsio
+    - cutlass
+    - dlpack
+    - libabseil
+    - libboost-headers  ${{ libboost_headers }}
+    - libjpeg-turbo
+    - libnvimgcodec-dev  ${{ libnvimgcodec }}
+    - libnvjpeg2k-dev  ${{ libnvjpeg2k }}
+    - libopencv
+    - libprotobuf
+    - libtiff
+    - pip
+    - protobuf
+    - pybind11
+    - python
+    - python-clang
+    - rapidjson
+    - setuptools
+    # Runtime deps from below, for stub generator
+    - astunparse ${{ astunparse }}
+    - gast ${{ gast }}
+    - six ${{ six }}
+    - dm-tree ${{ dm_tree }}
+    - packaging ${{ packaging }}
+    # Stubs autoformat themselves
+    - black  ${{ black }}
+  run:
+    - python
+    - astunparse ${{ astunparse }}
+    - gast ${{ gast }}
+    - six ${{ six }}
+    - dm-tree ${{ dm_tree }}
+    - packaging ${{ packaging }}
+    - ${{ pin_compatible('cuda-version', lower_bound='x', upper_bound='x') }}
+  run_constraints:
+    # These packages are dlopen'd and are already constrained by cuda-version
+    # FIXME: Consider a meta-package like "nvidia-dali-all" for convenience
+    # - if: match(cuda_compiler_version, "12.*")
+    #   then:
+    #     - libcufft
+    #     - libcufile
+    #     - libnpp
+    #     - libnvjpeg
+    - libboost-headers  ${{ libboost_headers }}
+    - libnvimgcodec0 ${{ libnvimgcodec }}
+tests:
+  - requirements:
+      run:
+        - pip
+    script:
+      - pip check
+      - python -c "import nvidia.dali"
+      - tfrecord2idx --help
+      - wds2idx --help
+about:
+  summary: A GPU-accelerated library for data loading and pre-processing to accelerate deep learning applications.
+  description: |
+    A GPU-accelerated library containing highly optimized building blocks and
+    an execution engine for data processing to accelerate deep learning training
+    and inference applications.
+  license: Apache-2.0 AND BSD-3-Clause AND BSD-2-Clause
+  license_file:
+    - LICENSE
```

Comment:
How about adding Acknowledgements.txt from DALI instead of all third_party/*? We aggregate all the necessary attributions there.

Reply:
https://github.com/conda-forge/nvidia-dali-python-feedstock/pull/34/files#r2095939803

Reply:
Thanks for the idea!